src/main/java/org/elasticsearch/index/query/FilteredQueryParser.java

```diff
@@ -45,70 +45,6 @@
 
     public static final String NAME = "filtered";
 
-    public static final FilterStrategy ALWAYS_RANDOM_ACCESS_FILTER_STRATEGY = new CustomRandomAccessFilterStrategy(0);
```

Comment:
To me this is still a mess. can we ignore all these options and return a BooleanQuery using FILTER clause? BooleanQuery is smarter than this one, for example it looks for approximations when considering bulk scoring and stuff.